playbooks/roles/download-and-verify/tasks/main.yml

```diff
@@ -1,41 +1,50 @@
 ---
-- name: "{{ project_signer }} signs the {{ project_name }} downloads. Import the correct GPG key."
-  command: gpg --keyserver {{ gpg_key_server_address }} --recv-keys {{ project_signing_key }}
-  register: gpg_recv_keys_result
-  changed_when: False
-  until: gpg_recv_keys_result | success
-  retries: 10
-  delay: 5
 
-- name: Register the GPG Key fingerprint output
-  command: gpg --fingerprint {{ project_signing_key }}
-  register: retrieved_fingerprint
-  changed_when: False
+# Import the GPG playbook vars
+- include_vars: "../../gpg/vars/main.yml"
 
-- name: Make sure the retrieved fingerprint perfectly matches the expected fingerprint
-  assert: { that: "project_expected_fingerprint in retrieved_fingerprint.stdout" }
-
-- name: Download the {{ project_name }} files and signatures
+- name: "Download the {{ project_name }} files"
   get_url:
-    url: "{{ item }}"
-    dest: "{{ project_download_location }}"
+    url: "{{ project_download_baseurl }}/{{ item.file }}"
+    dest: "{{ project_download_location }}/{{ item.file }}"
     owner: www-data
     group: www-data
     mode: 0644
-  with_items: "{{ project_download_urls }}"
+  with_items: "{{ project_download_files }}"
 
-- name: If necessary, rename signature files to match the GPG default of '.asc'
-  shell: for file in *.{{ signature_extension }}; do mv $file ${file%.*}.asc; done
-  args:
-    chdir: "{{ project_download_location }}"
-  when: signature_extension != "asc"
+- name: "Download the {{ project_name }} signatures"
+  get_url:
+    url: "{{ project_download_baseurl }}/{{ item.sig }}"
+    dest: "{{ project_download_location }}/{{ item.sig }}"
+    owner: www-data
+    group: www-data
+    mode: 0644
+  with_items: "{{ project_download_files }}"
 
-- name: Register GPG signature verification results
-  shell: for file in *.asc; do gpg --verify $file; done
-  args:
-    chdir: "{{ project_download_location }}"
+- name: "Verify the {{ project_name }} download signatures with the Streisand GPG keyring"
+  command: "gpgv2 --keyring {{ streisand_gpg_keyring }} {{ project_download_location }}/{{ item.sig }} {{ project_download_location }}/{{ item.file }}"
   register: gpg_verification_results
-  changed_when: False
+  with_items: "{{ project_download_files }}"
+
+- name: "Verify the {{ project_name }} download signature checks all passed"
```

Comment:
This seems a little spammy compared to other stuff. 

```
TASK [download-and-verify : Verify the OpenConnect download signature checks all passed] ******************************************
ok: [localhost] => (item={'_ansible_parsed': True, 'stderr_lines': [u'gpgv: Signature made Sun 07 Jan 2018 03:36:54 PM UTC using RSA key ID 9013B842', u'gpgv: Good signature from "Nikos Mavrogiannopoulos <nmav@gnutls.org>"', u'gpgv:                 aka "Nikos Mavrogiannopoulos <nmav@hushmail.com>"', u'gpgv:                 aka "Nikos Mavrogiannopoulos <n.mavrogiannopoulos@gmail.com>"'], '_ansible_item_result': True, u'end': u'2018-03-26 23:35:25.680773', '_ansible_no_log': False, u'stdout': u'', u'cmd': [u'gpgv2', u'--keyring', u'/root/.gnupg/streisand/pubring.gpg', u'/usr/local/src//ocserv-0.11.10.tar.xz.sig', u'/usr/local/src//ocserv-0.11.10.tar.xz'], u'rc': 0, 'item': {u'sig': u'ocserv-0.11.10.tar.xz.sig', u'file': u'ocserv-0.11.10.tar.xz'}, u'delta': u'0:00:00.009391', u'stderr': u'gpgv: Signature made Sun 07 Jan 2018 03:36:54 PM UTC using RSA key ID 9013B842\ngpgv: Good signature from "Nikos Mavrogiannopoulos <nmav@gnutls.org>"\ngpgv:                 aka "Nikos Mavrogiannopoulos <nmav@hushmail.com>"\ngpgv:                 aka "Nikos Mavrogiannopoulos <n.mavrogiannopoulos@gmail.com>"', u'changed': True, u'invocation': {u'module_args': {u'creates': None, u'executable': None, u'_uses_shell': False, u'_raw_params': u'gpgv2 --keyring /root/.gnupg/streisand/pubring.gpg /usr/local/src//ocserv-0.11.10.tar.xz.sig /usr/local/src//ocserv-0.11.10.tar.xz', u'removes': None, u'warn': True, u'chdir': None, u'stdin': None}}, 'stdout_lines': [], u'start': u'2018-03-26 23:35:25.671382', '_ansible_ignore_errors': None, 'failed': False}) => {
    "changed": false,
    "item": {
        "changed": true,
        "cmd": [
            "gpgv2",
            "--keyring",
            "/root/.gnupg/streisand/pubring.gpg",
            "/usr/local/src//ocserv-0.11.10.tar.xz.sig",
            "/usr/local/src//ocserv-0.11.10.tar.xz"
        ],
        "delta": "0:00:00.009391",
        "end": "2018-03-26 23:35:25.680773",
        "failed": false,
        "invocation": {
            "module_args": {
                "_raw_params": "gpgv2 --keyring /root/.gnupg/streisand/pubring.gpg /usr/local/src//ocserv-0.11.10.tar.xz.sig /usr/local/src//ocserv-0.11.10.tar.xz",
                "_uses_shell": false,
                "chdir": null,
                "creates": null,
                "executable": null,
                "removes": null,
                "stdin": null,
                "warn": true
            }
        },
        "item": {
            "file": "ocserv-0.11.10.tar.xz",
            "sig": "ocserv-0.11.10.tar.xz.sig"
        },
        "rc": 0,
        "start": "2018-03-26 23:35:25.671382",
        "stderr": "gpgv: Signature made Sun 07 Jan 2018 03:36:54 PM UTC using RSA key ID 9013B842\ngpgv: Good signature from \"Nikos Mavrogiannopoulos <nmav@gnutls.org>\"\ngpgv:                 aka \"Nikos Mavrogiannopoulos <nmav@hushmail.com>\"\ngpgv:                 aka \"Nikos Mavrogiannopoulos <n.mavrogiannopoulos@gmail.com>\"",
        "stderr_lines": [
            "gpgv: Signature made Sun 07 Jan 2018 03:36:54 PM UTC using RSA key ID 9013B842",
            "gpgv: Good signature from \"Nikos Mavrogiannopoulos <nmav@gnutls.org>\"",
            "gpgv:                 aka \"Nikos Mavrogiannopoulos <nmav@hushmail.com>\"",
            "gpgv:                 aka \"Nikos Mavrogiannopoulos <n.mavrogiannopoulos@gmail.com>\""
        ],
        "stdout": "",
        "stdout_lines": []
    },
    "msg": "All assertions passed"
}

TASK [download-and-verify : Verify the OpenConnect download signatures were from the correct keys] ****************
```

Reply:
Agreed! I meant to try and find a way to silence this super verbose `assert` output and then forgot about it. I'll take that as an item to clean-up.

Reply:
This turns out to be a real pain in the butt. There's an open upstream Ansible issue (https://github.com/ansible/ansible/issues/27124) about the verbosity of the `assert` task but unfortunately it seems like its stalled somewhere around what I'd consider WONTFIX.

I've [bumped that issue](https://github.com/ansible/ansible/issues/27124#issuecomment-376523736) to try and make the case that this is really unfortunate behaviour that should be fixed.

Reply:
It's unlikely I'll be able to come up with a clever solution to this in the next day or so and this PR fixes a bug multiple users have reported. In the interest of moving it forwards I added a `no_log` to each of the asserts and the output is much better now, e.g.:

```
TASK [download-and-verify : Verify the The Libreswan Project download signature checks all passed] ***
ok: [localhost] => (item=(censored due to no_log)) => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result"}
TASK [download-and-verify : Verify the The Libreswan Project download signatures were from the correct keys] ***
ok: [localhost] => (item=(censored due to no_log)) => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result"}
```

We'll have to disable `no_log` if the asserts fail to debug further.

@nopdotcom Does this resolve your review feedback?

Reply:
Yes, and you put way more effort into it than I thought! Thanks.